wire-ios-transport/Source/Background/BackgroundActivityFactory.swift

```diff
@@ -280,25 +278,27 @@ extension BackgroundActivityFactory {
 
     @objc
     func startTimer() {
-        if backgroundTaskTimer == nil {
-            backgroundTaskTimer = Timer.scheduledTimer(withTimeInterval: backgroundTaskTimeout,
-                                                       repeats: false,
-                                                       block: { [weak self] (timer) in
-                                                        self?.mainQueue.async { [weak self] in
-                                                            zmLog.safePublic("Handle expiration when the background task has timed out")
-                                                            self?.handleExpiration()
-                                                            timer.invalidate()
-                                                        }
-                                                       })
-        }
+        guard backgroundTaskTimer == nil else { return }
```

Comment:
```suggestion
        backgroundTaskTimer?.invalidate()
```

Reply:
Isn't this a different behaviour?
I understand it now as "if there is a backgroundTaskTimer set, don't start a new timer", but with the suggested change it would be "invalidate a running timer (if set) and start a new timer"